system-tests/management-api/management-api-test-runner/src/test/java/org/eclipse/edc/test/e2e/managementapi/AssetApiEndToEndTest.java

```diff
@@ -0,0 +1,150 @@
+import org.junit.jupiter.api.Test;
+
+import java.util.List;
+import java.util.Map;
+
+import static io.restassured.RestAssured.given;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.eclipse.edc.jsonld.spi.JsonLdKeywords.CONTEXT;
+import static org.eclipse.edc.jsonld.spi.JsonLdKeywords.ID;
+import static org.eclipse.edc.jsonld.spi.JsonLdKeywords.TYPE;
+import static org.eclipse.edc.jsonld.spi.JsonLdKeywords.VOCAB;
+import static org.eclipse.edc.jsonld.spi.PropertyAndTypeNames.EDC_ASSET_TYPE;
+import static org.eclipse.edc.spi.CoreConstants.EDC_NAMESPACE;
+import static org.hamcrest.Matchers.is;
+
+@EndToEndTest
+public class AssetApiEndToEndTest extends BaseManagementApiEndToEndTest {
+
+    private static final String TEST_ASSET_ID = "test-asset-id";
+    private static final String TEST_ASSET_CONTENTTYPE = "application/json";
+    private static final String TEST_ASSET_DESCRIPTION = "test description";
+    private static final String TEST_ASSET_VERSION = "0.4.2";
+    private static final String TEST_ASSET_NAME = "test-asset";
+
+    @Test
+    @Disabled(value = "Can enable once we have a unified transformer registry")
+    void getAssetById() {
+        //insert one asset into the index
+        controlPlane.getContext().getService(AssetIndex.class)
+                .accept(new AssetEntry(Asset.Builder.newInstance().id("test-asset").build(),
+                        DataAddress.Builder.newInstance().type("test-type").build()));
+
+        var body = baseRequest()
+                .get("/test-asset")
+                .then()
+                .statusCode(200)
+                .extract().body().as(new TypeRef<Map<String, Object>>() {
+                });
+
+        assertThat(body).isNotNull();
+        assertThat(body).containsKey("asset");
+        assertThat(body.get("asset")).isInstanceOf(Map.class);
+        Map<String, Object> asset = (Map<String, Object>) body.get("asset");
+        assertThat(asset.get(Asset.PROPERTY_ID).toString()).contains("test-asset");
+    }
+
+    @Test
+    @Disabled(value = "Can enable once we have a unified transformer registry")
+    void createAsset_shouldBeStored() {
+
+        var assetJson = Json.createObjectBuilder()
+                .add(CONTEXT, createContextBuilder().build())
+                .add(TYPE, EDC_ASSET_TYPE)
+                .add(ID, TEST_ASSET_ID)
+                .add("properties", createPropertiesBuilder().build())
+                .build();
+
+        var json = Map.of("asset", assetJson,
+                "dataAddress", Map.of("properties", Map.of("type", "test-type")));
+
+        var id = baseRequest()
+                .contentType(ContentType.JSON)
+                .body(json)
+                .post()
+                .then()
+                .log().ifError()
+                .statusCode(200)
+                .extract().body().as(IdResponseDto.class);
+
+        assertThat(id).isNotNull().extracting(IdResponseDto::getId).isEqualTo("test-asset-id");
```

Comment:
this assertion can be done through rest assured: `contentType(JSON).body("id", is("test-asset-id")`